src/openrct2/windows/_legacy.cpp

```diff
@@ -124,7 +124,7 @@ money32 place_provisional_track_piece(
     }
 }
 
-static std::tuple<bool, uint8_t> window_ride_construction_update_state_get_track_element()
+static std::tuple<bool, track_type_t> window_ride_construction_update_state_get_track_element()
```

Comment:
```suggestion
static std::optional<track_type_t> window_ride_construction_update_state_get_track_element()
```
Makes the usage of this function a bit more clear.

Reply:
I would suggest we leave that to a different PR. Keep this one purely focused on splitting the booster.

Reply:
I made it return only track_type_t to use TrackElemType::None as the 'false' value.

Reply:
this fails the CI so I dropped the commit